parcels/include/parcels.h

```diff
@@ -459,131 +461,82 @@ static inline StatusCode temporal_interpolation_structured_grid(type_coord x, ty
+			  tsrch, t0, t1, interp_method);
+  CHECKSTATUS(status);
+
+  if (grid->zdim == 1) {
+    // last param is a flag, which denotes that we only want the first timestep
+    // (rather than both)
+    status = getCell2D(f, xi[igrid], yi[igrid], ti[igrid], data2D, tii == 1); CHECKSTATUS(status);
   } else {
-    double t0 = grid->time[ti[igrid]];
-    status = search_indices(x, y, z, grid, &xi[igrid], &yi[igrid], &zi[igrid], &xsi, &eta, &zeta, gcode, ti[igrid], t0, t0, t0+1, interp_method); CHECKSTATUS(status);
-    if (grid->zdim==1){
-      status = getCell2D(f, xi[igrid], yi[igrid], ti[igrid], data2D, 1); CHECKSTATUS(status);
-    } else{
-      status = getCell3D(f, xi[igrid], yi[igrid], zi[igrid], ti[igrid], data3D, 1); CHECKSTATUS(status);
-    }
-    if ((interp_method == LINEAR) || (interp_method == CGRID_VELOCITY) || (interp_method == BGRID_VELOCITY) ||(interp_method == BGRID_W_VELOCITY)){
-      if ((interp_method == CGRID_VELOCITY) || (interp_method == BGRID_W_VELOCITY)){ // interpolate w
-        if (gridindexingtype == NEMO){
-          xsi = 1;
-          eta = 1;
-        }
-        else if (gridindexingtype == MITGCM){
-          xsi = 0;
-          eta = 0;
-        }
-        if (grid->zdim==1)
-          return ERROR;
-      }
-      else if (interp_method == BGRID_VELOCITY){
-        zeta = 0;
-      }    
-      if (grid->zdim==1){
-        status = spatial_interpolation_bilinear(xsi, eta, data2D[0], value); CHECKSTATUS(status);
-      }
-      else{
-        status = spatial_interpolation_trilinear(xsi, eta, zeta, data3D[0], value); CHECKSTATUS(status);
-      }
-    }
-    else if (interp_method == NEAREST){
-      if (grid->zdim==1){
-        status = spatial_interpolation_nearest2D(xsi, eta, data2D[0], value); CHECKSTATUS(status);
-      }
-      else {
-        status = spatial_interpolation_nearest3D(xsi, eta, zeta, data3D[0], value); CHECKSTATUS(status);
-      }
-    }
-    else if ((interp_method == CGRID_TRACER) || (interp_method == BGRID_TRACER)){
-      if (grid->zdim==1){
-        status = spatial_interpolation_tracer_c_grid_2D(data2D[0], value); CHECKSTATUS(status);
-      }
-      else {
-        status = spatial_interpolation_tracer_c_grid_3D(data3D[0], value); CHECKSTATUS(status);
-      }
-    }
-    else if (interp_method == LINEAR_INVDIST_LAND_TRACER){
-      if (grid->zdim==1){
-        status = spatial_interpolation_bilinear_invdist_land(xsi, eta, data2D[0], value); CHECKSTATUS(status);
-      }
-      else {
-        status = spatial_interpolation_trilinear_invdist_land(xsi, eta, zeta, data3D[0], value); CHECKSTATUS(status);
-      }
-    }
-    else {
-        return ERROR;    
-    }
-    return SUCCESS;
+    status = getCell3D(f, xi[igrid], yi[igrid], zi[igrid], ti[igrid], data3D, tii == 1); CHECKSTATUS(status);
+  }
+
+  // define a helper macro that will select the appropriate interpolation method
+  // depending on whether we need 2D or 3D
+#define INTERP(fn_2d, fn_3d)						\
```

Comment:
Why the back-slashes here and in the lines below? Do they do anything?

Reply:
C macros terminate at the end of the line, so we need continuation characters -- the whole backslashed block is the macro definition.

Reply:
Ah I see. Sorry for my ignorance on C; it's been a while since I've actively used it